internal/gatewayapi/route.go

```diff
@@ -390,16 +390,40 @@ func (t *Translator) processGRPCRouteRule(grpcRoute *GRPCRouteContext, ruleIdx i
 		}
 
 		if match.Method != nil {
-			if match.Method.Method != nil {
-				irRoute.HeaderMatches = append(irRoute.HeaderMatches, &ir.StringMatch{
-					Name:  ":method",
-					Exact: match.Method.Method,
-				})
-			}
-			/* TODO
-			if match.Method.Service != nil {
+			// GRPC's path is in the form of "/<service>/<method>"
+			//nolint:gocritic
+			switch GRPCMethodMatchTypeDerefOr(match.Method.Type, v1alpha2.GRPCMethodMatchExact) {
+			case v1alpha2.GRPCMethodMatchExact:
+				if match.Method.Service != nil && match.Method.Method != nil {
```

Comment:
this code can be simplified with 
```
if match.Method.Method != nil {
  pathMatch += XX
}
if match.Method.Service != nil {
  pathMatch += XX
}

irRoute.PathMatch = pathMatch
```

Reply:
or maybe it can't because of the regex needed for `Service` ?

Reply:
@arkodg we need to use regex if only method is specified, so this wouldn't work.